tests/utils.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import { getCopyOnEnterAttributes } from '../src/utils';
+import ModelTestEditor from '@ckeditor/ckeditor5-core/tests/_utils/modeltesteditor';
+
+describe( 'utils', () => {
+	describe( 'getCopyOnEnterAttributes()', () => {
+		it( 'filters attributes with copyOnEnter property', () => {
+			return ModelTestEditor.create()
+				.then( editor => {
+					const schema = editor.model.schema;
+
+					schema.register( 'foo', { inheritAllFrom: '$block' } );
```

Comment:
This line registers a schema object (not an attribute) which is a `block`, similar to `paragraph`.

AFAIR only the `schema.setAttributeProperties()` is required but to have this more elegant you could add:

```js
model.schema.extend( '$text', P allowAttributes: [ 'foo', 'bar', 'baz' ] } );
```

as this behavior is for text attributes only.